docs/containers/debug-node.md

```diff
@@ -124,10 +115,107 @@ Also note that the `debug` logger will not write logs unless enabled via the `DE
+
+For example, suppose the application logs the following message:
+
+```js
+function onListening() {
+  var addr = server.address();
+  var bind = typeof addr === 'string'
+    ? 'pipe ' + addr
+    : 'port ' + addr.port;
+  debug('Application has started on ' + bind);
+}
+```
+
+The corresponding `pattern` in the debug launch configuration (in `launch.json`) would be:
+
+```json
+{
+    "configurations": [
+        {
+            "name": "Docker Node.js Launch and Attach",
+            "type": "docker",
+            "request": "launch",
+            "preLaunchTask": "docker-run: debug",
+            "platform": "node",
+            "dockerServerReadyAction": {
+                "pattern": "Application has started on port (\\d+)"
+            }
+        }
+    ]
+}
+```
+
+### Configuring the application entry page
+
+By default, the Docker extension will open the "main" page of the browser (however that is determined by the application).  If the browser should be opened to a specific page, the `uriFormat` property of the [`dockerServerReadyAction`]((debug-common.md#dockerServerReadyAction-object-properties)) object of the debug launch configuration should be set to a Node.js format string, with one string token that indicates where the port should be substituted.
+
+The corresponding `uriFormat` in the debug launch configuration (in `launch.json`) to open the `about.html` page instead of the main page would be:
+
+```json
+{
+    "configurations": [
+        {
+            "name": "Docker Node.js Launch and Attach",
+            "type": "docker",
+            "request": "launch",
+            "preLaunchTask": "docker-run: debug",
+            "platform": "node",
+            "dockerServerReadyAction": {
+                "uriFormat": "http://localhost:%s/about.html"
+            }
+        }
+    ]
+}
+```
+
+## Mapping Docker container source files to the local workspace
+
+By default, the Docker extension assumes the application source files in the running Docker container are located in an `/usr/src/app` folder, and the debugger then maps those files back to the root of the opened workspace, in order to translate breakpoints from the container back to Visual Studio Code.
+
+If the application source files are in a different location (e.g. different Node.js frameworks have different conventions), either within the Docker container or within the opened workspace, then one or both of the `localRoot` and `remoteRoot` properties of the [`node`](debug-node.md#node-object-properties) object of the debug launch configuration should be set the root source locations within the workspace and the Docker container, respectively.
```

Comment:
We don't use "e.g." - replace with "for example, "

Reply:
Updated.